stackslib/src/net/api/postblock_proposal.rs

```diff
@@ -194,6 +196,136 @@ impl NakamotoBlockProposal {
             })
     }
 
+    /// Check to see if a block builds atop the highest block in a given tenure.
+    /// That is:
+    /// - its parent must exist, and
+    /// - its parent must be as high as the highest block in the given tenure.
+    pub(crate) fn check_block_builds_on_highest_block_in_tenure(
+        chainstate: &StacksChainState,
+        tenure_id: &ConsensusHash,
+        parent_block_id: &StacksBlockId,
+    ) -> Result<(), BlockValidateRejectReason> {
+        let Some(highest_header) = NakamotoChainState::get_highest_known_block_header_in_tenure(
```

Comment:
This function get_highest_known_block_header_in_tenure says "Do not use in consensus code"...should the function check_block_builds_on_highest_block_in_tenure also have this warning since it calls it?

Reply:
Sure -- done.